.github/workflows/test-generated-project.yml

```
@@ -0,0 +1,58 @@
+name: Test Generated Project
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  TERRAFORM_VERSION: "1.5.5"
+  TFSEC_VERSION: "v1.28.1"
+
+jobs:
+  test:
+    name: Run Tests Generated Project
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Cache Node npm
+        id: cache-nodemodules
+        uses: actions/cache@v3
```

we can upgrade to 4 version here
```suggestion
        uses: actions/cache@v4
```

---

Updated in [add9bd2](https://github.com/nimblehq/infrastructure-templates/pull/276/commits/add9bd2b01d1cad45d64496943de5dc827cc3d14)